src/models/stores/persistent-ui/persistent-ui.ts

```diff
@@ -222,8 +222,31 @@ export const PersistentUIModelV2 = types
      *
      * @param doc a non curriculum document
      */
-    openResourceDocument(doc: DocumentModelType, user?: UserModelType, sortedDocuments?: SortedDocuments) {
-      const navTab = getNavTabOfDocument(doc, user)  || "";
+    openResourceDocument(
+      doc: DocumentModelType,
+      appConfig: AppConfigModelType,
```

Comment:
Ideally we'd setup our stores so that they can depend on each other. So in this case the PersistentUI could say it needs the appConfig and then it would have access to it. This way openResourceDocument wouldn't need to be passed the appConfig all of the time. 

There are examples of this in `stores.ts` already where one store is passed to another store, but it isn't a good approach. The approach doesn't prevent circular references, or make sure a store depending on another store is always created in the right order. 

It doesn't make sense build that system in this PR, I just wanted to point it out incase you come back to the stores work in the future.